source/server/overload_manager_impl.cc

```diff
@@ -413,6 +413,17 @@ OverloadManagerImpl::OverloadManagerImpl(Event::Dispatcher& dispatcher, Stats::S
     const auto& name = action.name();
     const auto symbol = action_symbol_table_.get(name);
     ENVOY_LOG(debug, "Adding overload action {}", name);
+
+    // Validate that this is a well known overload action.
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.overload_manager_error_unknown_action")) {
+      auto& well_known_actions = OverloadActionNames::get().WellKnownActions;
+      if (std::find(well_known_actions.begin(), well_known_actions.end(), name) ==
+          well_known_actions.end()) {
+        throw EnvoyException(absl::StrCat("Unknown Overload Manager Action ", name));
+      }
+    }
```

Comment:
You're obviously way more familiar with this than I am, so I apologize if this is an obvious question. If overload actions were like normal Envoy extensions, we would not want to have a list of known actions because that would prevent someone from adding a custom extension in their downstream repository. Is that a concern here? Is it possible to add a new OverloadAction? Maybe this is what you mean by "difficult to extend without upstream changes" in the PR description?

Maybe we just want to flesh this out in the PR description a bit?

Reply:
Thanks, added some more details.

cc @nezdolik